pkg/api/types.go

```diff
@@ -3023,6 +3023,10 @@ type NodeList struct {
 type NamespaceSpec struct {
 	// Finalizers is an opaque list of values that must be empty to permanently remove object from storage
 	Finalizers []FinalizerName
+
+	// NetworkPolicy describes the use of NetworkPolicy objects in this namespace. If not specified, then default policies will be applied and NetworkPolicy objects in the Namespace will be ignored
+	// +optional
+	NetworkPolicy *NamespaceNetworkPolicy
```

Comment:
@jbeda @bgrant0607 regarding layering of the system, network policy has (at least until now) seemed "not core".  However, we need a place to record per namespace defaults that is clear and readable to end API consumers.  If not on Namespace, where would we put it?  A `NamespaceNetworkPolicy` that is cluster scoped and has the same name as the namespace?  Using a label selector?  How far do we go to avoid layering violations and keep the core simple, but still preserve usability?

An annotation is the least objectionable option that keeps it on Namespace.  Most of the other possible alternatives are more complex.